src/date-input/date-utils.tsx

```diff
@@ -67,5 +96,11 @@ export function formatDateISO(date: Date | undefined) {
 }
 
 export function formatDate(date: Date) {
-	return Intl.DateTimeFormat().format(date);
+	const formattedDate = Intl.DateTimeFormat().format(date);
+	// Check if is a supported date format
+	const parsedDate = parseDate(formattedDate);
+	if (parsedDate && parsedDate.valueOf() === date.valueOf()) {
+		return formattedDate;
+	}
+	return formatDateISO(date);
```

Comment:
Warning: this function uses `String.prototype.padStart`, which is not supported in IE11. I'll raise an issue and this can handled in a separate PR.

Reply:
This is now fixed in `master`.